spock-specs/src/test/groovy/org/spockframework/smoke/ast/condition/ExceptionConditionsAstSpec.groovy

```diff
@@ -28,18 +28,26 @@ public java.lang.Object foobar() {
 }
 
 public void $spock_feature_0_0() {
-    java.lang.Object foobar
-    this.getSpecificationContext().setThrownException(null)
     try {
-        foobar = this.foobar()
-    }
-    catch (java.lang.Throwable $spock_ex) {
-        this.getSpecificationContext().setThrownException($spock_ex)
+        java.lang.Object foobar
+        this.getSpecificationContext().setThrownException(null)
+        try {
+            org.spockframework.runtime.SpockRuntime.callEnterBlock(this.getSpecificationContext(), new org.spockframework.runtime.model.BlockInfo(org.spockframework.runtime.model.BlockKind.WHEN, []))
+            foobar = this.foobar()
+            org.spockframework.runtime.SpockRuntime.callExitBlock(this.getSpecificationContext(), new org.spockframework.runtime.model.BlockInfo(org.spockframework.runtime.model.BlockKind.WHEN, []))
```

Comment:
💬 this won't be called, fixing this makes for some awkward interactions between the individual AST transformations,

Reply:
I tried moving the blockListener logic where the blocks are written back in `SpecRewriter#L388`, however this didn't work for some cases, e.g. `cleanup` as it already replaced all Blocks with an anonymous block.